recipe/meta.yaml

```
@@ -36,43 +36,47 @@ requirements:
     - python
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }} # [(ep_variant == "cuda")]
     - cmake
-    - git  # [not win]
+    - git       # [not win]
     - ninja     # [not win]
-    - patch    # [unix]
-    - m2-patch # [win]
+    - patch     # [unix and (ep_variant == "cuda")]
+    - m2-patch  # [win]
   host:
     - python
     - pip
     - wheel
     - flake8
     - packaging
     - python-flatbuffers
-    - zlib 1.2.13
-    - zstd {{ zstd }}
-    - numpy {{ numpy }}
-    - pybind11 2.10.4
-    - setuptools
-    - wheel
+    - zlib 1
+    - numpy 2.0  # [py<313]
+    - numpy 2.1  # [py>=313]
+    - pybind11 2
     # sympy, pytest and gmock are required to run tests
     - sympy
     - pytest
     - gmock
     # GPU requirements
-    - cudatoolkit {{ cudatoolkit }}*  # [(ep_variant == "cuda") and (linux and x86_64)]
-    - cudnn {{ cudnn }}*              # [(ep_variant == "cuda") and (linux and x86_64)]
-    - cupti {{ cudatoolkit }}*        # [(ep_variant == "cuda") and (linux and x86_64)]
+    - cuda-nvcc {{ cuda_compiler_version }}        # [(ep_variant == "cuda")]
+    - cuda-nvrtc-dev {{ cuda_compiler_version }}   # [(ep_variant == "cuda")]
+    - cuda-cudart-dev {{ cuda_compiler_version }}  # [(ep_variant == "cuda")]
+    # Needed for CUDA profiling.                   # [(ep_variant == "cuda")]
+    - cuda-cupti-dev {{ cuda_compiler_version }}   # [(ep_variant == "cuda")]
+    - cudnn 9.1.1.17*                              # [(ep_variant == "cuda")]
+    - libcublas-dev                                # [(ep_variant == "cuda")]
+    - libcufft-dev                                 # [(ep_variant == "cuda")]
+    - libcurand-dev                                # [(ep_variant == "cuda")]
+    - libcusparse-dev                              # [(ep_variant == "cuda")]
   run:
+    - __cuda  # [linux and (ep_variant == "cuda")]
     - coloredlogs
     - packaging
     - protobuf
     - python
     - python-flatbuffers
     - sympy
     - {{ pin_compatible('numpy') }}
```

```suggestion
    - numpy >= 1.21.6,<3
```
Especially for backward compatibility.

---

There are currently build issues with python < 3.13 when using numpy > 2.0. Would we prefer to pin >=1.21.6,<2.1 or just rely on building with numpy 2.0 at this point? What's the benefit to possibly building with numpy < 2.0?

---

This package should be buildable for python 3.13 (built against numpy 2.1 for 3.13). Could you try again?

---

Especially if build issues are due to missing packages, this can be addressed.

---

That is how it was tested originally. The host section has numpy set as `- numpy 2.1  # [py>=313]`. Do we want a 3.13 variant built at this time? I did not make one when I manually built the GPU variants.

---

Yes, and this also needs to be added to abs.yaml
```
build_env_vars:
  ANACONDA_ROCKET_ENABLE_PY313 : yes
```